src/v/raft/heartbeat_manager.cc

```diff
@@ -273,7 +275,7 @@ ss::future<> heartbeat_manager::do_heartbeat(node_heartbeat&& r) {
 
 void heartbeat_manager::process_reply(
   model::node_id n,
-  absl::btree_map<raft::group_id, follower_request_meta> groups,
+  const absl::node_hash_map<raft::group_id, follower_request_meta>& groups,
```

Comment:
whats this change for?

Reply:
i just wanted to accommodate to what we agreed previously in slack that the `node_hash_map` is better suited in the cases when we do not need ordering as it provides enough prevention from memory fragmentation.